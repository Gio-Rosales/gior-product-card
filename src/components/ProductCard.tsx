import React, { createContext } from 'react';
import { useProduct } from '../hooks/useProduct';

import { ProductContextProps, ProductCardProps } from '../interfaces/interfaces';

import styles from '../styles/styles.module.css';

export const ProductContext = createContext({} as ProductContextProps);
const { Provider } = ProductContext;

export const ProductCard = (
  {
    product,
    children,
    className,
    style,
    onChange,
    value,
    initialValues
  }: ProductCardProps) => {
  const { counter, isMaxCountReached, maxCount, increaseBy, reset } = useProduct({ onChange, product, value, initialValues });

  // console.log('counter', counter);
  return (
    <Provider value={{
      product,
      counter,
      increaseBy,
      maxCount
    }}>
      <div className={`${styles.productCard} ${className}`} id={product.id} style={style}>
        {children({
          count: counter,
          isMaxCountReached,
          maxCount,
          product,

          increaseBy,
          reset
        })}
      </div>
    </Provider>
  )
};
