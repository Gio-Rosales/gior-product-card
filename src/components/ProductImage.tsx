import React, { useContext } from 'react';
import { ProductContext } from './ProductCard';

import { ProductImageProps } from '../interfaces/interfaces';

import styles from '../styles/styles.module.css';
import noImage from '../assets/no-image.jpg';

export const ProductImage = ({ img, className, style }: ProductImageProps) => {
  const { product } = useContext(ProductContext);
  let imageToShow: string;

  if (img) {
    imageToShow = img;
  } else if (product.img) {
    imageToShow = product.img;
  } else {
    imageToShow = noImage;
  }

  return (
    <img
      src={imageToShow}
      alt="Product"
      className={`${styles.productImg} ${className}`}
      style={style}
    />
  );
};