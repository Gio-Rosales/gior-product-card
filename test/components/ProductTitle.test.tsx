import React from 'react';
import renderer from 'react-test-renderer';
import { ProductTitle, ProductCard } from '../../src/components';
import { productWOImage } from '../data/products';

describe('ProductTitle', () => {
  it('should show correctly with a custom title', () => {
    const wrapper = renderer.create(
      <ProductTitle title='Custom Product' className='custom-class' />
    );

    console.log(wrapper.toJSON());
    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should show the component with the product\'s name', () => {
    const wrapper = renderer.create(
      <ProductCard product={productWOImage}>
        {
          () => (
            <ProductTitle />
          )
        }
      </ProductCard>
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });
});
