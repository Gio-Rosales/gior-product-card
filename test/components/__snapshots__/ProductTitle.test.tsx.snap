// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductTitle should show correctly with a custom title 1`] = `
<span
  className="productDescription custom-class"
>
  Custom Product
</span>
`;

exports[`ProductTitle should show the component with the product's name 1`] = `
<div
  className="productCard undefined"
  id="1"
>
  <span
    className="productDescription undefined"
  >
    Coffee Mug - Without Image
  </span>
</div>
`;
